### PR TITLE
Adding various devtools such as eslint, prettier, and vscode settings

### PR DESCRIPTION
**Pull request by itstang**
- Migrated to use latest version of Yarn from npm
- Added recommended extensions for VSCode
- Added Prettier and enabled autosave
- Added .editorconfig file for base editor configurations
- Added eslint and used airbnb base linter rules
- Formatted existing code to fix linting errors

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    "editor.formatOnType": true
+  },
+  "[typescriptreact]": {
+    "editor.defaultFormatter": "esbenp.prettier-vscode",
+    "editor.formatOnSave": true,
+    "editor.formatOnType": true
+  },
+  "[json]": {
+    "editor.defaultFormatter": "esbenp.prettier-vscode",
+    "editor.formatOnSave": true,
```
Comment (KeijiBranshi) [inline .vscode/settings.json]: any issues with just doing this across the board as opposed to individual basis?

File: package.json [hunk 2]
```diff
@@ -24,16 +31,25 @@
...
     "@types/react-dom": "^16.9.5",
-    "react": "^16.13.0"
+    "@typescript-eslint/eslint-plugin": "^2.22.0",
+    "@typescript-eslint/parser": "^2.22.0",
+    "eslint-config-prettier": "^6.10.0",
+    "eslint-plugin-prettier": "^3.1.2",
+    "prettier": "^1.19.1",
+    "react": "^16.13.0",
+    "react-dom": "^16.13.0"
   }
```
Comment (KeijiBranshi) [inline package.json]: do we want any pre-commit hooks for lint/prettier as well?

File: package.json [hunk 3]
```diff
@@ -24,16 +31,25 @@
...
     "html-webpack-plugin": "^3.2.0",
     "html-webpack-template": "^6.2.0",
-    "react-dom": "^16.13.0",
     "ts-loader": "^6.2.1",
     "typescript": "^3.8.3",
     "webpack": "^4.41.6",
     "webpack-cli": "^3.3.11"
   },
   "dependencies": {
     "@types/react-dom": "^16.9.5",
```
Comment (KeijiBranshi) [inline package.json]: nit: devDependencies?

File: package.json [hunk 4]
```diff
@@ -24,16 +31,25 @@
...
-    "react": "^16.13.0"
+    "@typescript-eslint/eslint-plugin": "^2.22.0",
+    "@typescript-eslint/parser": "^2.22.0",
+    "eslint-config-prettier": "^6.10.0",
+    "eslint-plugin-prettier": "^3.1.2",
+    "prettier": "^1.19.1",
```
Comment (KeijiBranshi) [inline package.json]: nit: devDependencies?

File: .vscode/settings.json (hunk 1 above)
Comment (itstang) [reply .vscode/settings.json]: I figure these are something that would be pretty universal enough across the board

File: package.json (hunk 2 above)
Comment (itstang) [reply package.json]: I think that would be useful, but I didn't want to add too much in one PR

**Pull request merged**